src/middleware/index.ts

```diff
@@ -0,0 +1,13 @@
+import { UiState } from '../types';
+
+export interface MiddlewareDefinition {
+  onStateChange(options: { state: UiState }): void;
+  subscribe(): void;
+  unsubscribe(): void;
+}
+
+export type Middleware = ({
+  instantSearchInstance: InstantSearch,
+}) => MiddlewareDefinition;
+
+export { createRouter } from './createRouter';
```

Comment:
Would have gone with simply `router` instead of `createRouter` here. I found this more natural when using it with `this.use()`: `this.use(router(options))`

WDYT?

Reply:
Sure! I didn't know if there were any naming convention for middleware.

Reply:
Note that the word `router` will be repeated:

```js
router({ router, stateMapping })
```